ios/brave-ios/Sources/Brave/Frontend/Browser/SearchResultAdClickedInfoBar.swift

```diff
@@ -0,0 +1,171 @@
+  static let learnMoreUrl = "https://support.brave.com/hc/en-us/articles/360026361072-Brave-Ads-FAQ"
+}
+
+class SearchResultAdClickedInfoBar: Toast, UITextViewDelegate {
+  let tabManager: TabManager
+
+  init(tabManager: TabManager) {
+    self.tabManager = tabManager
+
+    super.init(frame: .zero)
+
+    self.tapDismissalMode = .outsideTap
+
+    self.clipsToBounds = true
+    self.addSubview(
+      createView(
+        Strings.searchResultAdsClickedInfoBarTitle + " "
+      )
+    )
+
+    self.toastView.backgroundColor = SearchResultAdClickedInfoBarUX.toastBackgroundColor
+
+    self.toastView.snp.makeConstraints { make in
+      make.left.right.height.equalTo(self)
+      self.animationConstraint =
+        make.top.equalTo(self).offset(SearchResultAdClickedInfoBarUX.toastHeight).constraint
+    }
+
+    self.snp.makeConstraints { make in
+      make.height.equalTo(SearchResultAdClickedInfoBarUX.toastHeight)
+    }
+  }
+
+  required init?(coder aDecoder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  fileprivate func createView(
+    _ labelText: String
+  ) -> UIView {
+    let horizontalStackView = UIStackView()
+    horizontalStackView.axis = .horizontal
+    horizontalStackView.alignment = .center
+    horizontalStackView.spacing = SearchResultAdClickedInfoBarUX.toastPadding
+
+    let label = UITextView()
+    label.textAlignment = .left
+    label.textColor = .white
+    label.font = SearchResultAdClickedInfoBarUX.toastLabelFont
+    label.backgroundColor = SearchResultAdClickedInfoBarUX.toastBackgroundColor
+    label.isEditable = false
+    label.isScrollEnabled = false
+    label.isSelectable = true
+    label.delegate = self
+
+    let learnMoreText = Strings.learnMore.withNonBreakingSpace
+    let attributes: [NSAttributedString.Key: Any] = [
+      .foregroundColor: UIColor.white,
+      .font: SearchResultAdClickedInfoBarUX.toastLabelFont,
+    ]
+
+    let linkAttributes: [NSAttributedString.Key: Any] = [
+      .font: SearchResultAdClickedInfoBarUX.toastLabelFont,
+      .foregroundColor: UIColor.white,
+      .underlineStyle: 1,
+    ]
+    label.linkTextAttributes = linkAttributes
+
+    let nsLabelAttributedString = NSMutableAttributedString(
+      string: labelText,
+      attributes: attributes
+    )
+    let nsLinkAttributedString = NSMutableAttributedString(
+      string: learnMoreText,
+      attributes: linkAttributes
+    )
+
+    if let url = URL(string: SearchResultAdClickedInfoBarUX.learnMoreUrl) {
+      let learnMoreRange = NSRange(location: 0, length: learnMoreText.count)
+      nsLinkAttributedString.addAttribute(.link, value: url, range: learnMoreRange)
+      nsLabelAttributedString.append(nsLinkAttributedString)
+      label.isUserInteractionEnabled = true
+    }
+    label.attributedText = nsLabelAttributedString
+
+    horizontalStackView.addArrangedSubview(label)
+
+    if let buttonImage = UIImage(braveSystemNamed: "leo.close") {
```

Comment:
Lose this if-let entirely and just set the image directly, add an assert if you particularly want to check for non-nil but it should always be there

Reply:
Fixed